### PR TITLE
Home work 9

### PR DESCRIPTION
**Pull request by UnitOmega13**

### CODE REVIEW

File: src/NewHashMap.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+			resizeOfTable();
+		}
+		Node<K, V> newNode = new Node<>(key, value);
+		Node<K, V> currentNode = table[hashKeyIndex(key, table.length)];
+		if (currentNode == null) {
+			table[hashKeyIndex(key, table.length)] = newNode;
+		} else if (currentNode.keyValue == key) {
+			currentNode.itemValue = value;
+		} else {
+			while (currentNode.hasNext()) {
```
Comment (SevostyanovGerman) [inline src/NewHashMap.java]: write comment here, i can't understand that

**Review by SevostyanovGerman — CHANGES_REQUESTED** [review]

File: src/NewHashMap.java (hunk 1 above)
Comment (UnitOmega13) [reply src/NewHashMap.java]: I added comments to the project

File: src/NewHashMap.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	public void put(K key, V value) {
+		int occupancy = (int) (table.length * TABLE_MAX_PERCENT_OCCUPANCY);
+		if (sizeOfTable == 0) {                                                //Check if hashmap empty, create new
+			table = new Node[TABLE_SIZE];                                      //with size = TABLE_SIZE
+		} else if (occupancy <= TABLE_SIZE) {                                  //If number of elements more than
+			resizeOfTable();                                                   //75% of hashmap size we resize our
+		}                                                                      //hashmap with method resizeOfTable.
+		Node<K, V> newNode = new Node<>(key, value);                           //Create node with key/value that we
+																			   // want to put in hashmap
+		Node<K, V> currentNode = table[hashKeyIndex(key, table.length)];       //Create node to check does node exist
```
Comment (SevostyanovGerman) [inline src/NewHashMap.java]: cs

File: src/NewHashMap.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+																			   //with this key
+		if (currentNode == null) {                                             //If does not exist then put new node
+			table[hashKeyIndex(key, table.length)] = newNode;                  //put in a cell with a given key
+		} else if (currentNode.keyValue == key) {                              //If exist than replace value in cell
+			currentNode.itemValue = value;
+		}else {
```
Comment (SevostyanovGerman) [inline src/NewHashMap.java]: doesn't work

Comment (SevostyanovGerman) [reply src/NewHashMap.java]: and ==

**Review by SevostyanovGerman — CHANGES_REQUESTED** [review]

File: src/NewHashMap.java (hunk 3 above)
Comment (UnitOmega13) [reply src/NewHashMap.java]: fixed

**Review by SevostyanovGerman — APPROVED** [review]

**Pull request closed without merging**